src/AspNet.Security.OAuth.SuperOffice/Implementation/DefaultSuperOfficeConfigurationManager.cs

```diff
@@ -0,0 +1,101 @@
+﻿/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+using System;
+using System.Net.Http;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.Extensions.Logging;
+using Microsoft.IdentityModel.Tokens;
+
+namespace AspNet.Security.OAuth.SuperOffice.Implementation
+{
+    internal sealed class DefaultSuperOfficeConfigurationManager : SuperOfficeAuthenticationConfigurationManager
+    {
+        private readonly ISystemClock _clock;
+        private readonly ILogger _logger;
+
+        private string _publicKey = string.Empty;
+        private DateTimeOffset _reloadKeysAfter;
+
+        public DefaultSuperOfficeConfigurationManager(
+            ISystemClock clock,
```

Comment:
Lacks the relevant `[NotNull]` attributes the rest of the providers use (and in the provider this code was copied from).